File: C++ Code/Basics/Others/Prime.cpp
// Q. Check wheather a number given by the user is prime or not
#include<iostream>
using namespace std;
int main(){
    int a;
    cout<<"Enter a number : ";
    cin>>a;
    if(a==1){
        cout<<"1 is neither a prime nor a composite number.";
        return 0;
    }
    bool b=true;
    for(int i=2;i<a;i++){
        if(a%i==0){
            b=false;
            break;
        }
    }
    if(b)
        cout<<"The given number "<<a<<" is prime.";
    else    
        cout<<"The given number "<<a<<" is composite.";
    return 0;
}

File: Java Code/Basics/Other/Prime.java
// WAP to find whether a number given by user is prime or not.

import java.util.*;

public class Prime{
  public static void main(String[] args){
    Scanner sc=new Scanner(System.in);
    System.out.print("Enter the Number : ");
    int n=sc.nextInt();
    if(n<2){
      System.out.println(n+" is neither Prime nor Composite ..");
      return ;
    }
    boolean IsPrime=true;
    for(int i=2;i<=Math.sqrt(n);i++){
      if(n%i==0){
        IsPrime=false;
        break;
      }
    }
    if(IsPrime)
      System.out.println(n+" is Prime Number.");
    else
        System.out.println(n+" is Composite Number.");
  }
}